### PR TITLE
add tests for GMG with elasticity

### PR DESCRIPTION
**Pull request by naliboff**
This PR adds two tests that use elasticity in combination with the GMG solver. While this combination is currently permitted, there are no tests that explicitly test to make sure it works. However, @zjiaqi2018 @tjhei and I did test the GMG + elasticity combination a few months ago and confirmed it worked for models with mesh deformation and/or surface tractions. This PR does not include tests for either of the latter two features (still disabled with GMG), but does ensure GMG is able to run with elasticity.
 
* [X] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).
* [X] I have tested my new feature locally to ensure it is correct.
* [X] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [ ] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

**Comment by tjhei** [comment]
Awesome, thank you. I have two questions:
- are they confirmed to give the same results as AMG?
- I assume you might the resolution tiny and reduced the number of timesteps to make it run fast. Would it make sense to also include the original/ bigger files? I think those would be a nice fit for ./examples/ (which doesn't exist yet), but see #4505

**Comment by naliboff** [comment]
Odd, an unrelated test failed. Restarting the testers to see if this is repeatable.

**Comment by bangerth** [comment]
It seemed to work this time!

File: tests/viscoelastic_bending_beam_gmg.prm [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# This test checks whether the viscoelastic bending beam benchmark
+# run successfully. In particular, this benchmark is a good test to
```
Comment (tjhei) [inline tests/viscoelastic_bending_beam_gmg.prm:2]:
```suggestion
# runs successfully. In particular, this benchmark is a good test to
```

**Comment by naliboff** [comment]
AMG versus GMG results comparisons for viscoelastic stress build-up problem after 20 time steps:

AMG
```
   Postprocessing:
     Compositions min/max/mass: 9.152e+07/9.152e+07/9.152e+17 // -9.152e+07/-9.152e+07/-9.152e+17 // -88.79/42.77/-1.13e+10
     Temperature min/avg/max:   293 K, 293 K, 293 K
     RMS, max velocity:         0.0258 m/year, 0.0445 m/year
```

GMG
```
   Postprocessing:
     Compositions min/max/mass: 9.152e+07/9.152e+07/9.152e+17 // -9.152e+07/-9.152e+07/-9.152e+17 // -143.1/107.6/-3.27e+08
     Temperature min/avg/max:   293 K, 293 K, 293 K
     RMS, max velocity:         0.0258 m/year, 0.0445 m/year
```

The viscoelastic stress_xx and stress_yy components are identical, but the are small variations in the stress_xy values. However, the stress_xy values are ~ 7 orders of magnitude smaller than the stress_xx and stress_yy values, so I'm inclined to chalk the differences up numerical roundoff between the AMG and GMG solvers.

**Comment by bangerth** [comment]
Yes, compared to the size of the diagonal entries of the stress, these changes seem negligible.

Out of curiosity, what is the difference in runtime between the GMG and AMG solvers for this problem?

**Comment by naliboff** [comment]
@bangerth - I'm not sure about the timing, as I ran these on my Mac (in debug mode) where I'm not sure if the precompiled deal.II build has native optimizations turned on. I will check this on a linux machine where I know GMG is working correctly.

**Comment by bangerth** [comment]
No need to spend any time on it. I was just curious!

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Did you update the test output before merging?

**Comment by naliboff** [comment]
I did not have a chance to update it, are the tests in other PRs passing even after #4713 was merged?

**Comment by tjhei** [comment]
no, master is broken now: https://github.com/geodynamics/aspect/commits/main
I will fix it...

**Pull request merged**